### ISSUE TITLE
[Tarea 2] Coffee - acción take

### ISSUE THREAD

**Issue by benjaminlillo** [IIC2613-Inteligencia-Artificial-2021-2/Syllabus#110]
Hola, no entiendo cuál es el propósito de la acción take en la parte de coffee single agent. Por lo que entiendo, cuando se prepara un trago con prepare(drink) el hecho de que se haya creado se refleja en que el robot ahora tiene ese trago en la mano, y no veo otra forma de declarar que ese trago ha sido preparado (antes pensaba que con el fluent has(Room, Drink) podía "guardarlo en la cocina", pero ahora se llama delivered y no tendría sentido). Entonces, si cuando se prepara un trago este se va directamente al robot, no entiendo por qué sería necesaria la acción take(drink). Tampoco la veo necesaria si se refiere a llevar un trago a una habitación, pues ya existe la acción deliver.

**Comment by Dietr1ch** [comment]
> No entiendo cuál es el propósito de la acción `take` en la parte de coffee single agent. Por lo que entiendo, cuando se prepara un trago con `prepare(drink)` el hecho de que se haya creado se refleja en que el robot ahora tiene ese trago en la mano

No, no lo tiene en la mano altiro. Puedes pensar que está usando una máquina de café y que queda listo allí, pero que en vez de introducir la noción de máquina de café simplemente decimos que la cocina ahora tiene café.

> no veo otra forma de declarar que ese trago ha sido preparado (antes pensaba que con el fluent `has(Room, Drink)` podía "guardarlo en la cocina", pero ahora se llama `delivered` y no tendría sentido). Entonces, si cuando se prepara un trago este se va directamente al robot, no entiendo por qué sería necesaria la acción `take(drink)`. Tampoco la veo necesaria si se refiere a llevar un trago a una habitación, pues ya existe la acción `deliver`.

`delivered(Room, Drink)` es un objetivo, que se cumple cuando la pieza eventualmente tiene café servido (`has(Room, Drink)`). No es el mismo fluente porque algunos querían más libertad, pero lo que interesa es que finalmente se cumpla `has(Room, Drink)` en el tiempo final para todos los requests de `delivered`.
